hs_collection_resource/page_processors.py

```diff
@@ -0,0 +1,62 @@
+from mezzanine.pages.page_processors import processor_for
+
+from hs_core import page_processors
+from hs_core.views import add_generic_context
+from hs_core.models import BaseResource
+from hs_access_control.models import PrivilegeCodes
+
+from .models import CollectionResource
+
+@processor_for(CollectionResource)
+def landing_page(request, page):
+    content_model = page.get_content_model()
+    edit_resource = page_processors.check_resource_mode(request)
+
+    user = request.user
+    if user.is_authenticated():
+        # get a list of resources with effective OWNER privilege
+        owned_resources = user.uaccess.get_resources_with_explicit_access(PrivilegeCodes.OWNER)
+        # get a list of resources with effective CHANGE privilege
+        editable_resources = user.uaccess.get_resources_with_explicit_access(PrivilegeCodes.CHANGE)
+        # get a list of resources with effective VIEW privilege
+        viewable_resources = user.uaccess.get_resources_with_explicit_access(PrivilegeCodes.VIEW)
+
+        owned_resources = list(owned_resources)
+        editable_resources = list(editable_resources)
+        viewable_resources = list(viewable_resources)
+        discovered_resources = list(user.ulabels.my_resources)
+
+        user_all_accessible_resource_list = (owned_resources + editable_resources + \
+                                             viewable_resources + discovered_resources)
```

Comment:
@zhiyuli @pkdash This block of code to get user_all_accessible_resource_list is repetitive of the code block here https://github.com/hydroshare/hydroshare/blob/develop/hs_core/views/__init__.py#L482..L517 in my_resources() view function. In fact, this code block does not exclude obsoleted resources from the list as my_resources() view function does, which proves it is important we make this as a common function to be called by both collection page processor and my_resources() page processor view function. You can move the block in my_resources() page processor to a common utility function, e.g., get__all_accessible_resource_list(), which can be called in both places.